drivers/sdhc/sdhci_common.c

```diff
@@ -437,7 +572,11 @@ static void sdhci_prepare_dma(struct sdhci_common *sdhci_ctx, struct sdhc_data *
 
 	dma_addr = z_mem_phys_addr(data->data);
 	LOG_DBG("sdhc:req: SDHCI_DMA_ADDRESS %08lx", dma_addr);
-	sys_write32(dma_addr, sdhci_ctx->reg_base + SDHCI_DMA_ADDRESS);
+	if (sdhci_ctx->dma_mode == SDHC_DMA_SDMA) {
+		sys_write32(dma_addr, sdhci_ctx->reg_base + SDHCI_DMA_ADDRESS);
```

Comment:
This comment isn't related to this PR.
Please follow the flow from section `3.7.2.2 Using SDMA` in the `SD Host Controller Simplified Specification Version 3.00`.

I mean it is needed to reprogram the new system address at every page boundary:
```
DMA had disadvantage that DMA Interrupt generated at every
page boundary disturbs CPU to reprogram the new system address. This SDMA algorithm forms a
performance bottleneck by interruption at every page boundary.
```

Reply:
It's generated not at the **page** boundary, but at value configured in "Block Size Register" -> "SDMA Buffer Boundary". It's set to 512K and:
- such max transfers covers now all Zephyr can do
- in adma PR check added to xfer to block bigger sdma xfers